src/main/kotlin/ru/driics/aitrade/service/PromptFormatterService.kt

```
@@ -0,0 +1,97 @@
+                num.setScale(5, RoundingMode.HALF_UP).stripTrailingZeros().toPlainString()
+            else ->
+                num.setScale(8, RoundingMode.HALF_UP).stripTrailingZeros().toPlainString()
+        }
+    }
+
+    fun formatScientific(num: BigDecimal?): String {
+        if (num == null || num == BigDecimal.ZERO) return "0"
+
+        val abs = num.abs()
+        return if (abs < BigDecimal("0.0001")) {
+            String.format("%.2e", num.toDouble())
```

detekt (1.23.8)

[warning] String.format("%.2e", num.toDouble()) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)